include/swift/AST/TypeCheckRequests.h

```diff
@@ -78,13 +81,75 @@ void simple_display(
 
 void simple_display(llvm::raw_ostream &out, ASTContext *ctx);
 
+/// Emulates the following enum with associated values:
```

Comment:
This seems like a lot of new code for just the new `~BitwiseCopyable` syntax. Can you instead find a way to generalize the existing logic for `~Copyable`?

Reply:
I think having a different representation for this case--the case of a protocol that isn't conformed to by all types by default (i.e. unlike `Copyable`) but is inferred automatically for some types--makes sense.